samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -311,6 +321,68 @@ private StreamPartitionCountMonitor getPartitionCountMonitor(Config config, Syst
       });
   }
 
+  private Optional<InputStreamRegexMonitor> getInputRegexMonitor(Config config, SystemAdmins systemAdmins) {
+
+    // if input regex monitor is not enabled return empty
+    if (new JobConfig(config).getMonitorInputRegexFrequency() <= 0) {
```

Comment:
Consider exposing `isStreamMonitoringEnabled` as a convenience API on the JobConfig class?

Reply:
done